nixpkgs-overlays/default.nix

```diff
@@ -0,0 +1,15 @@
+self: super:
+
+let
+  inherit (self) lib;
+  inherit (import ../dep/gitignore.nix { inherit lib; }) gitignoreSource;
+in {
+  obeliskCleanSource = src:
+    # WARNING: The order of application here seems to matter a great deal to
+    # how quickly `ghcid` is able to reload changes. As a rule of thumb,
+    # always apply `gitignoreSource` first.
+    # See https://github.com/obsidiansystems/obelisk/pull/666 and related.
+    lib.cleanSource (gitignoreSource src);
```

Comment:
I do have doubts about exposing this here, but otherwise the overlays are not self contained. Could call it `_obeliskCleanSource` or something to indicate it is subject to change.